crates/next-core/src/render_from_node/render_static.rs

```diff
@@ -0,0 +1,259 @@
+use std::collections::HashMap;
+
+use anyhow::{bail, Context, Result};
+use mime::TEXT_HTML_UTF_8;
+use serde::{Deserialize, Serialize};
+use turbo_tasks::{primitives::StringVc, Value};
+use turbo_tasks_fs::{File, FileContent, FileSystemEntryType, FileSystemPathVc};
+use turbopack::{transition::TransitionsByNameVc, ModuleAssetContextVc};
+use turbopack_core::{
+    asset::{Asset, AssetContent, AssetContentVc, AssetVc},
+    chunk::ChunkingContextVc,
+    context::AssetContextVc,
+    environment::{EnvironmentIntention, EnvironmentVc, ExecutionEnvironment, NodeJsEnvironment},
+    reference_type::{EntryReferenceSubType, ReferenceType},
+    source_asset::SourceAssetVc,
+};
+use turbopack_dev_server::html::DevHtmlAssetVc;
+use turbopack_ecmascript::{
+    chunk::EcmascriptChunkPlaceablesVc, EcmascriptInputTransformsVc, EcmascriptModuleAssetType,
+    EcmascriptModuleAssetVc,
+};
+use turbopack_node::{
+    evaluate::{evaluate, JavaScriptValue},
+    get_intermediate_asset, get_renderer_pool, trace_stack, NodeJsOperation,
+};
+
+use super::{
+    issue::RenderingIssue, RenderDataVc, RenderResult, RenderStaticIncomingMessage,
+    RenderStaticOutgoingMessage,
+};
+use crate::{
+    embed_js::next_asset,
+    next_server::{get_build_module_options_context, get_build_resolve_options_context},
+};
+
+/// Renders a module as static HTML in a node.js process.
+#[turbo_tasks::function]
+pub async fn render_static(
+    path: FileSystemPathVc,
+    project_root: FileSystemPathVc,
+    module: EcmascriptModuleAssetVc,
+    runtime_entries: EcmascriptChunkPlaceablesVc,
+    fallback_page: DevHtmlAssetVc,
+    chunking_context: ChunkingContextVc,
+    intermediate_output_path: FileSystemPathVc,
+    data: RenderDataVc,
+) -> Result<AssetContentVc> {
+    let ecma_chunk = module.as_evaluated_chunk(chunking_context, Some(runtime_entries));
+    for referenced_asset in ecma_chunk.references().await?.iter() {
+        for asset in referenced_asset
+            .resolve_reference()
+            .primary_assets()
+            .await?
+            .iter()
+        {
+            // let mut inner_asset = HashMap::new();
```

Comment:
TODO, create a new  `EcmascriptModuleAssetVc` with the, `inner_asset`,  the `inner_asset` contains the processed CSS by postcss.